sdk/storage/azblob/blockblob/client_test.go

```diff
@@ -5486,3 +5487,39 @@ func TestRequestIDGeneration(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, requestIdMatch, true)
 }
+
+type serviceVersionTest struct{}
+
+// newServiceVersionTestPolicy returns a policy that checks the x-ms-version header
+func newServiceVersionTestPolicy() policy.Policy {
+	return &serviceVersionTest{}
+}
+
+func (m serviceVersionTest) Do(req *policy.Request) (*http.Response, error) {
+	const versionHeader = "x-ms-version"
+
+	currentVersion := map[string][]string(req.Raw().Header)[versionHeader]
+	if currentVersion[0] != generated.ServiceVersion {
+		return nil, fmt.Errorf(currentVersion[0] + " service version doesn't match expected version: " + generated.ServiceVersion)
+	}
+	return req.Next()
```

Comment:
You don't really have to return req.Next(). That will make the test actually hit the service wont it?

What you can also do is just call GetProperties, that way we don't have to create data unnecessarily.

Reply:
That's true, I changed it to a fake response being returned.